framework/src/org/checkerframework/common/value/util/Range.java

```diff
@@ -109,23 +112,33 @@ public boolean isNothing() {
     /**
      * Converts a this range to a 32-bit integral range.
      *
-     * <p>If this range is too wide, i.e., wider than the full range of the Integer class, return
-     * INT_EVERYTHING.
+     * <p>If {@link #IGNORE_OVERFLOW} is true and one of the bounds is outside the Integer range,
+     * then that bound is set to the bound of the Integer range.
+     *
+     * <p>If {@link #IGNORE_OVERFLOW} is false and this range is too wide, i.e., wider than the full
+     * range of the Integer class, return INT_EVERYTHING.
      *
-     * <p>If the bounds of this range are not representable as 32-bit integers, convert the bounds
-     * to Integer type in accordance with Java overflow rules, e.g., Integer.MAX_VALUE + 1 is
-     * converted to Integer.MIN_VALUE.
+     * <p>If {@link #IGNORE_OVERFLOW} is false and the bounds of this range are not representable as
+     * 32-bit integers, convert the bounds to Integer type in accordance with Java overflow rules,
+     * e.g., Integer.MAX_VALUE + 1 is converted to Integer.MIN_VALUE.
      */
     public Range intRange() {
-        if (this.isWiderThan(integerWidth)) {
-            return INT_EVERYTHING;
+        if (this.isNothing()) {
+            return NOTHING;
+        }
+        if (IGNORE_OVERFLOW) {
+            return new Range(Math.max(from, Integer.MIN_VALUE), Math.min(to, Integer.MAX_VALUE));
         } else {
```

Comment:
In general, I often don't use an `else` clause after a return statement, both because it causes deep indentation and because that indentation makes me keep in mind what else may have occurred, rather than knowing that everything that has gone before is irrelevant.  I see that you did this for the `isNothing` special case above.